backend/hitas/models/apartment.py

```diff
@@ -129,6 +133,30 @@ def is_new(self) -> bool:
 
         return no_first_purchase_or_in_the_future and no_first_sale_or_in_the_future
 
+    def sell_by_date(self):
+        sell_by_dates: set[datetime.date] = set()
+
+        for ownership in self.ownerships.all():
+            for cos in chain(ownership.conditions_of_sale_new.all(), ownership.conditions_of_sale_old.all()):
```

Comment:
We are starting to have a bit of these related model accessions, that assume the related objects are cached.
In the future might want to consider creating tests for these, to make sure they are actually already fetched, and dont cause unnecessary db queries. What do you think? 🤔

Reply:
I made a context manager test tool that can be used to count the database queries happening in its context. Here is an example of it's usage:

```python
# Expect that calling this endpoint will make 5 database queries
with count_queries(5):
    response = api_client.get(reverse("hitas:apartment-list"))
```

Let me know what you think. There is also the tedious part of going through every test and adding this context manager to them...

Reply:
I didn't think this would have been a part of this ticket, but ok sure 😆 

> There is also the tedious part of going through every test and adding this context manager to them
And also figuring out what is the correct amount of queries for the context 🙂